File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxCurr = 0;
        int maxSum = INT_MIN;
        
        for(int i=0; i<nums.size(); i++){
            maxCurr += nums[i];
            
            if(maxSum < maxCurr) maxSum = maxCurr;
            
            if(maxCurr < 0) maxCurr = 0;
        }
        
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        largest_sum = float('-inf')
        curr_sum = 0
        
        for num in nums:
            curr_sum += num     
            largest_sum = max(curr_sum, largest_sum)
            
            if curr_sum < 0:
                curr_sum=0
        
        return largest_sum